src/RequestForm.js

```diff
@@ -247,25 +243,13 @@ class RequestForm extends React.Component {
   }
 
   onChangeFulfilment(e) {
-    const {
-      defaultDeliveryAddressTypeId,
-      defaultServicePointId,
-    } = this.state;
-    const { change } = this.props;
-
-    const deliverySelected = e.target.value === fulfilmentTypeMap.DELIVERY;
+    const selectedFullfillmentPreference = e.target.value;
 
     this.setState({
-      selectedDelivery: deliverySelected,
+      selectedDelivery: this.isDeliverySelected(selectedFullfillmentPreference),
+    }, () => {
+      this.updateRequestPreferencesFields(selectedFullfillmentPreference);
```

Comment:
There's no need to pass `selectedFullfillmentPreference` to `updateRequestPreferencesFields` since all the needed data is stored in `state` which can be accessed from `updateRequestPreferencesFields`